### PR TITLE
Modernize a bit

### PR DESCRIPTION
**Pull request by Goooler**
Just all things.

### CODE REVIEW

**Comment by Goooler** [comment]
Version catalog migration has been tracked by https://github.com/thundernest/k-9/issues/6514.

File: .editorconfig
Comment (cketti) [inline .editorconfig:8]: There are a couple of places where we use trailing whitespace and removing it would break the code. I'm not generally opposed to this. But it should only be added after the code has been changed. Until then, please remove this line.

File: app/k9mail/build.gradle
Comment (cketti) [inline app/k9mail/build.gradle:97]: This line was kept intentionally to make it very explicit that minification is not enabled. Please don't remove it.

File: build.gradle
Comment (cketti) [inline build.gradle:59]: ❌ This has practical implications and it's easy to end up with code that won't run on older Android versions.

File: settings.gradle
Comment (cketti) [inline settings.gradle]: ❌ This might be fine if we had a dedicated build engineer. But we don't. So I'd like us to use a conservative build config. That means no preview features (unless there's a very good reason).

File: gradle.properties
Comment (cketti) [inline gradle.properties:12]: ❌ It looks like this is an experimental feature.

File: gradle.properties
Comment (cketti) [inline gradle.properties:4]: This value was intentionally low so people with hardware that is not very powerful can still contribute. If you have a suggestion on how to use a low default value, but have an easy way to override it for CI, I'm all ears.

File: gradle.properties
Comment (cketti) [inline gradle.properties:15]: It looks like this commit mixes unrelated changes. Please don't do that.

File: settings.gradle
Comment (cketti) [inline settings.gradle:14]: Is there a reason to put this here and not inside `build.gradle`?

File: settings.gradle
Comment (cketti) [inline settings.gradle:6]: ❌ I don't want us to use this on CI. We certainly don't want everyone who builds the app to upload data by default!

File: settings.gradle
Comment (cketti) [inline settings.gradle]: ❌ No preview features.

File: app/k9mail/build.gradle
Comment (cketti) [inline app/k9mail/build.gradle:65]: Why?

**Review by cketti — CHANGES_REQUESTED** [review]
Please try to limit the scope of pull requests. Especially if you can't know whether your changes are welcome.

The downside of "just all things" is that now there's a ton of unrelated things we have to talk about in one pull request.

File: build.gradle
Comment (Goooler) [reply build.gradle:59]: AGP will desugar them.

File: settings.gradle
Comment (Goooler) [reply settings.gradle]: Maybe we can add a todo here?

File: gradle.properties
Comment (Goooler) [reply gradle.properties:12]: Yep

File: app/k9mail/build.gradle
Comment (Goooler) [reply app/k9mail/build.gradle:65]: Use the default release signing by default, if not will make the debug signing as the fallback.

File: build.gradle
Comment (cketti) [reply build.gradle:59]: As far as I can tell, only if you also use `coreLibraryDesugaring`. We've been holding off because a) we don't really need it for our code and b) it didn't work with commons-io because of https://issuetracker.google.com/issues/160484830

The last time I checked AGP didn't use an r8 version that includes a fix for the issue mentioned above.

File: settings.gradle
Comment (cketti) [reply settings.gradle]: The problem isn't that we'll forget about removing this line. The issue is that we have better things to do than debugging failing builds. Not using experimental features reduces the risk of running into build issues.

File: build.gradle
Comment (Goooler) [reply build.gradle]: 84f9fe1f07a117e546e254dbef6d0fe407aeb92b

Comment (Goooler) [reply build.gradle]: 84f9fe1f07a117e546e254dbef6d0fe407aeb92b

File: app/k9mail/build.gradle
Comment (Goooler) [reply app/k9mail/build.gradle]: d56eb8d1ba748d1d2c5e6bc153b30e10c989d9ef

File: settings.gradle
Comment (Goooler) [reply settings.gradle]: 31ddac19c312266521636e1f4f7a8dc1f0cf01f7

File: gradle.properties
Comment (Goooler) [reply gradle.properties]: 31ddac19c312266521636e1f4f7a8dc1f0cf01f7

File: gradle.properties
Comment (Goooler) [reply gradle.properties:4]: 31ddac19c312266521636e1f4f7a8dc1f0cf01f7

File: gradle.properties
Comment (Goooler) [reply gradle.properties]: 31ddac19c312266521636e1f4f7a8dc1f0cf01f7

File: .editorconfig
Comment (Goooler) [reply .editorconfig]: ed578a6c6348fb9ff7e9cc78a2a771b5f06b3f89

File: app/k9mail/build.gradle
Comment (Goooler) [reply app/k9mail/build.gradle]: ed578a6c6348fb9ff7e9cc78a2a771b5f06b3f89

**Comment by Goooler** [comment]
You can squash this PR after approval.

File: settings.gradle
Comment (cketti) [reply settings.gradle]: I'd rather not use this at all.

**Review by cketti — APPROVED** [review]
Thanks :+1:

**Pull request merged**
